File: php/program/Mahasiswa.php
<?php

include_once("Human.php");
class Mahasiswa extends Human{
    protected $nim;
    protected $prodi;
    protected $fakultas;
    protected $imgPath;

    public function __construct($id, $name, $gender ,$nim, $prodi, $fakultas, $nik){
        $this->id = $id;
        $this->name = $name;
        $this->nim = $nim;
        $this->prodi = $prodi;
        $this->fakultas = $fakultas;
        $this->imgPath = "default.jpg";
        $this->gender = $gender;
        $this->nik = $nik;
    }

    //Setter
    public function setNim($input){
        $this->nim = $input;
    }
    public function setProdi($input){
        $this->prodi = $input;
    }
    public function setFakultas($input){
        $this->fakultas = $input;
    }
    public function setImgPath($input){
        $this->imgPath;
    }

    // Getter
    public function getNim(){
        return $this->nim;
    }
    public function getProdi(){
        return $this->prodi;
    }
    public function getFakultas(){
        return $this->fakultas;
    }
    public function getImgPath(){
        return $this->imgPath;
    }
}
?>

File: cpp/program/Mahasiswa.cpp
#pragma once

#include <iostream>
#include <string>
#include "Human.cpp"
#include "Academy.cpp"

class Mahasiswa : public Human, public Academy
{

private:
    std::string nim;
    std::string prodi;
    std::string fakultas;

public:
    Mahasiswa(
        std::string newName = "", 
        std::string newNim = "",
        std::string newProdi = "",
        std::string newFac = "",
        std::string newNik = "",
        GENDER newGender = GENDER::ATTACK_HELICOPTER,
        std::string newUniversity = "",
        std::string newUnivMail = "")
        {
            this->name = newName;
            this->nim = newNim;
            this->prodi = newProdi;
            this->fakultas = newFac;
            this->nik = newNik;
            this->gender = newGender;
            this->university = newUniversity;
            this->univ_mail = newUnivMail;
        }

    ~Mahasiswa(){}

    //Setter
    void setNim(std::string newNim)
    {
        this->nim = newNim;
    }
    void setProdi(std::string newProdi)
    {
        this->prodi = newProdi;
    }
    void setFac(std::string newFac)
    {
        this->fakultas = newFac;
    }

    //getter
    std::string getNim()
    {
        return this->nim;
    }
    std::string getProdi()
    {
        return this->prodi;
    }
    std::string getFac()
    {
        return this->fakultas;
    }

};